systemlayer/executionrequests.go

```diff
@@ -8,32 +8,58 @@ import (
 	pilosa "github.com/featurebasedb/featurebase/v3"
 )
 
+const (
+	keepLastRequests int = 2000
+	truncateTextAt   int = 4096
+)
+
 // ExecutionRequests is an internal struct that keeps a list of sql execution requests
 // this data allows visbility into queries that have been run and are running
 type ExecutionRequests struct {
 	sync.RWMutex
-	requests map[string]*pilosa.ExecutionRequest
+	requestsList []string
+	requests     map[string]*pilosa.ExecutionRequest
 }
 
 // Ensure type implements interface.
 var _ pilosa.ExecutionRequestsAPI = (*ExecutionRequests)(nil)
 
 func NewExecutionRequestsAPI() *ExecutionRequests {
 	return &ExecutionRequests{
-		requests: make(map[string]*pilosa.ExecutionRequest),
+		requestsList: make([]string, 0),
```

Comment:
This will allocate the requestsList once up front, rather than reallocating as it grows up to its max size. Not a big deal either way, but also makes it explicit that this is the size we expect it to be.
```suggestion
		requestsList: make([]string, 0, keepLastRequests),
```

Reply:
actually nevermind... because of the way we're appending below we're going to keep needing to reallocate. We'd have to implement a true ring buffer tracking the current index to avoid that.

Reply:
yeah agree